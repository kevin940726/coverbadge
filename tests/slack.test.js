import fetch from 'isomorphic-fetch';
import {
  formatMessage,
  sendSlackWebhook,
} from '../services/slack';

jest.mock('isomorphic-fetch');

describe('formatMessage', () => {
  it('should display correct info when decreased', () => {
    expect(formatMessage(66, 55)).toMatchSnapshot();
  });

  it('should display correct info when increased', () => {
    expect(formatMessage(55, 66)).toMatchSnapshot();
  });

  it('should display correct info when hit 100%', () => {
    expect(formatMessage(55, 100)).toMatchSnapshot();
  });

  it('should display correct info when remained', () => {
    expect(formatMessage(55, 55)).toMatchSnapshot();
  });

  it('should display correct info when remained and hit 100%', () => {
    expect(formatMessage(100, 100)).toMatchSnapshot();
  });
});

describe('sendSlackWebhook', () => {
  afterEach(() => {
    fetch.mockClear();
  });

  it('should call fetch with parameters', async () => {
    await sendSlackWebhook('slack', 50, 100);

    expect(fetch.mock.calls).toMatchSnapshot();
  });

  it('should format the text if have other options', async () => {
    await sendSlackWebhook('slack', 50, 100, {
      username: 'usernmame',
      project: 'project',
      vcs: 'github',
    });

    expect(fetch.mock.calls).toMatchSnapshot();
  });

  it('should further format the text if have branch', async () => {
    await sendSlackWebhook('slack', 50, 100, {
      username: 'usernmame',
      project: 'project',
      vcs: 'github',
      branch: 'branch',
    });

    expect(fetch.mock.calls).toMatchSnapshot();
  });
});
