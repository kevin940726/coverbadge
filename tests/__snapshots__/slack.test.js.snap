// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PR should handle link error 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"Coverage increased (*+50.00%*) to *100%*. (<https://github.com/kevin940726/coverbadge/pull/1|#1>)\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`PR should handle regex error 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"Coverage increased (*+50.00%*) to *100%*. (<https://github.com/kevin940726/coverbadge/pull/2|#2>)\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`PR should process multiple prs 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"Coverage increased (*+50.00%*) to *100%*. (<https://github.com/kevin940726/coverbadge/pull/1|#1>,<https://github.com/kevin940726/coverbadge/pull/2|#2>)\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`PR should process one pr 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"Coverage increased (*+50.00%*) to *100%*. (<https://github.com/kevin940726/coverbadge/pull/2|#2>)\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`formatMessage should display correct info when decreased 1`] = `
Object {
  "emoji": ":warning:",
  "text": "Coverage decreased (*-11.00%*) to *55%*.",
}
`;

exports[`formatMessage should display correct info when hit 100% 1`] = `
Object {
  "emoji": ":100:",
  "text": "Coverage increased (*+45.00%*) to *100%*.",
}
`;

exports[`formatMessage should display correct info when increased 1`] = `
Object {
  "emoji": ":tada:",
  "text": "Coverage increased (*+11.00%*) to *66%*.",
}
`;

exports[`formatMessage should display correct info when remained 1`] = `
Object {
  "emoji": ":bookmark:",
  "text": "Coverage remained the same at *55%*.",
}
`;

exports[`formatMessage should display correct info when remained and hit 100% 1`] = `
Object {
  "emoji": ":100:",
  "text": "Coverage remained the same at *100%*.",
}
`;

exports[`sendSlackWebhook should call fetch with parameters 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"Coverage increased (*+50.00%*) to *100%*.\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`sendSlackWebhook should format the text if have other options 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"<https://github.com/usernmame/project|[usernmame/project]> Coverage increased (*+50.00%*) to *100%*.\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`sendSlackWebhook should further format the text if have branch 1`] = `
Array [
  Array [
    "slack",
    Object {
      "body": "{\\"username\\":\\"coverbadge\\",\\"icon_emoji\\":\\":100:\\",\\"text\\":\\"<https://github.com/usernmame/project|[usernmame/project]> Coverage increased (*+50.00%*) to *100%*. on <https://github.com/usernmame/project/tree/branch|branch>\\"}",
      "headers": Object {
        "Content-Type": "application/json",
      },
      "method": "POST",
    },
  ],
]
`;
